airflow/hooks/jdbc_hook.py

```diff
@@ -25,64 +27,54 @@ class JdbcHook(BaseHook):
     :type sql: string or string pointing to a template file. File must have
         a '.sql' extensions.
     """
+
+
+    conn_name_attr = 'jdbc_conn_id'
+    default_conn_name = 'jdbc_default'
+    supports_autocommit = True
+
     def __init__(
-            self, jdbc_url, jdbc_driver_name, jdbc_driver_loc,host=None, login=None,
-            psw=None, db=None, port=None, extra=None, conn_id=None):
-        self.jdbc_driver_loc = jdbc_driver_loc
-        self.jdbc_driver_name = jdbc_driver_name
-
-        if (conn_id is None):
-            self.host = host
-            self.login = login
-            self.psw = psw
-            self.db = db
-            self.port = port
-            self.extra = extra
-        else:
-            conn = self.get_connection(conn_id)
-            self.host = conn.host
-            self.login = conn.login
-            self.psw = conn.password
-            self.db = conn.schema
-            self.port = conn.port
-            self.extra = conn.extra
-
-
-        self.jdbc_url = jdbc_url.format(self.host, self.port, self.db, self.extra)
+            self, *args, **kwargs):
+
+        super(JdbcHook,self).__init__(*args,**kwargs)
+
```

Comment:
I think there's no need for an `__init__` anymore as `DbApiHook ` should work just fine. 

You can move the `get_connection` call into `get_conn` and pass the info straight into the `connect` call.